src/credentials/loginManager.ts

```diff
@@ -21,13 +21,19 @@ export class LoginManager {
     private readonly defaultCredentialsRegion = 'us-east-1'
     private readonly credentialsStore: CredentialsStore = new CredentialsStore()
 
-    public constructor(private readonly awsContext: AwsContext) {}
+    public constructor(
+        private readonly awsContext: AwsContext,
+        public readonly recordAwsSetCredentialsFn: typeof recordAwsSetCredentials = recordAwsSetCredentials
+    ) {}
 
     /**
      * Establishes a Credentials for the Toolkit to use. Essentially the Toolkit becomes "logged in".
      * If an error occurs while trying to set up and verify these credentials, the Toolkit is "logged out".
+     *
+     * @param credentialsProviderId
+     * @param passive  If true, this was _not_ a user-initiated action.
      */
-    public async login(credentialsProviderId: CredentialsProviderId): Promise<void> {
+    public async login(passive: boolean, credentialsProviderId: CredentialsProviderId): Promise<void> {
```

Comment:
Consider making these parameters into a destructured object. Reading the actual function calls (eg: `await loginManager.login(true, loginCredentialsId)`), it is not apparent what the bool values represent.

Reply:
done